cmd/dagger/flags.go

```diff
@@ -285,7 +286,12 @@ func (v *directoryValue) Get(ctx context.Context, dag *dagger.Client, modSrc *da
 	// POSIX "portable filename character set":
 	// https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap03.html#tag_03_282
 	path, viewName, _ := strings.Cut(path, ":")
+	path, err = expandHomeDir(path)
+	if err != nil {
+		return nil, fmt.Errorf("failed to expand home directory: %w", err)
+	}
 	path = filepath.ToSlash(path) // make windows paths usable in the Linux engine container
+
 	return modSrc.ResolveDirectoryFromCaller(path, dagger.ModuleSourceResolveDirectoryFromCallerOpts{
 		ViewName: viewName,
 	}).Sync(ctx)
```

Comment:
May I ask why directory value need sync while file value is not?

Reply:
Ahh, this is so that if the path doesn't exist we can error here - instead of later, after we've done a bunch of work.

But yes, this is weird, we'd like to get rid of it at some point I think.